internal/common/db.go

```diff
@@ -341,6 +341,21 @@ func GetInfraEnvFromDBWhere(db *gorm.DB, where ...interface{}) (*InfraEnv, error
 	return &infraEnv, nil
 }
 
+func ResetAutoAssignRoles(db *gorm.DB, onClusters interface{}) (int, error) {
+	if db == nil {
+		return 0, nil
+	}
+
+	reply := db.Model(&models.Host{}).Where("role = ?", models.HostRoleAutoAssign).
```

Comment:
Since we are expecting the monitor to work after this, I think it will be a good idea to set also trigger_monitor_timestamp field